storage/pc-ata/ata.c

```diff
@@ -624,11 +638,63 @@ int ata_init(void)
 		return -ENOMEM;
 	}
 
-	if (ata_initbus((void *)(ATA1_BASE | 0x1), (void *)(ATA1_CTRL | 0x1), bus1) < 0)
+	for (i = 0; i < sizeof(pci_devClasses) + 1; i++) {
+		if (i == 0) {
```

Comment:
Way to remove this `if` (which kinda violates MISRA):
- init ata1Base etc before the loop
- break from loop on success
- if not success, then do the pctl stuff
- ???
- profit

Reply:
I think that organizing a loop in the prepare-then-init fashion is a bit more readable than the inverse, especially when there's has to be some way of skipping the init part if platformctl needs to be retried (i.e. when 0x0101 class device not present, but 0x0105 is). Currently this skip is realized by doing a `continue`, in inverse it may not be so clean

I wonder whether trying the initbuses for the common ports before the loop wasn't the best of all worlds :p

Reply:
Ok, I see, this trick with continue is colliding with my idea. I think you're right that the original idea was better.

Reply:
Ok, I reverted to the original idea